doc/sphinx.md

```diff
@@ -0,0 +1,208 @@
+Sphinx
+======
+
+## 1. Introduction
+
+This module manages the build of the documentation. The input files are C++,
+Python, and Markdown. In order to process this we use a couple of
+off-the-shelf softwares (see the list below).
+
+* Doxygen the C++ api documentation parser,
+* Breathe a sphinx extension that parse the doxygen xml output into restructured text files,
+* recommonmark a sphinx extension parsing markdown files.
+* sphinx-apidoc the Python api documentation parser,
+* Sphinx the documentation renderer,
+
+If anything seems fuzzy (it is a rather long and tedious pipeline),
+please let me know via posting an issue
+`here <https://github.com/machines-in-motion/mpi_cmake_modules/issues>`_.
+
+
+## 2. Advanced explanation on the tools
+
+In order to build the documentation we need to setup the following tools:
+- [Doxygen](http://www.doxygen.nl/) the C++ api documentation parser,
+- [Breathe](https://breathe.readthedocs.io/en/latest/) a sphinx extension that
+    parse the doxygen xml output into restructured text files,
+- [recommonmark](https://recommonmark.readthedocs.io/en/latest/) a sphinx
+    extension parsing markdown files.
+- [sphinx-apidoc](http://www.sphinx-doc.org/en/master/man/sphinx-apidoc.html)
+    the Python api documentation parser,
+- [Sphinx](http://www.sphinx-doc.org/en/master/) the documentation renderer,
+
+### 2.1 Doxygen
+
+In order to execute to generate the C++ API documentation we use the
+Doxygen tool.
+We wrote a `Doxyfile`, used to parameter Doxyygen, to notably:
+
+- Output the files in the `_build/docs/doxygen` folder with the
+  `OUTPUT_DIRECTORY` parameter. 
+- Generate a list of xml files containing the API documentation setting
+the `GENERATE_XML` to `YES`.
+
+The Makefile looks at the `Doxyfile` in `doc_config_files/doxygen/` and 
+CMake configure the `Doxyfile.in` from `cmake/doxygen/`.
+
+### 2.2 Breathe
+
+This tool is a module of sphinx that parse the Doxygen xml output.
+Breathe provide two import tools:
+
+- An API that allow you to reference to the object from the Doxygen xml
+  output.
+- An executable `breathe-apidoc` that generates automatically the C++ API
+  into ReStructed files.
+
+In order to use it we need to add a couple of line in the `config.py`
```

Comment:
Is `config.py` referring to the Sphinx config?  If so, isn't that file called `conf.py`?

Reply:
This also appears multiple times in the following text.